src/pages/BooksList.jsx

```diff
@@ -0,0 +1,75 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import "../styles/BooksList.css";
+
+function BooksList() {
+  const [booksData, setBooksData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const navigate = useNavigate();
+
+  const getListOfBooks = async () => {
+    try {
+      const res = await axios.post(
+        "https://api-preprod.lelivrescolaire.fr/graph",
+        {
+          query:
+            "query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}",
+        },
+        {
+          headers: {
+            "Content-Type": "application/json; charset=utf-8",
+          },
+        }
+      );
+      return res.data;
+    } catch (err) {
+      console.error("Error:", err.response ? err.response.data : err.message);
+
+      return null;
+    }
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const booksList = await getListOfBooks();
+      if (booksList) {
+        setBooksData(booksList.data.viewer.books.hits);
+      } else {
+        console.log("Erreur lors de la récupération des données.");
+      }
+      setLoading(false);
+    };
+
+    fetchData();
+  }, []);
+
+  const handleBookClick = (book) => {
+    if (book.valid) {
+      navigate(`/list/${book.id}`, { state: { book } });
```

Comment:
When passing data between screens, it's useful to avoid possible discrepancies. Here, I don't know if the `book` is serialized to the URL or passed in some other way, but the user might edit (or bookmark) the URL, expecting to be able to view the book. Without the state, this would not work.

This is related to one of Alex's comments regarding the fact that the list of books is reloaded whenever the home page is rendered. How would you avoid this?